src/rust/src/rdataframe/mod.rs

```diff
@@ -442,7 +444,57 @@ impl DataFrame {
             .map_err(polars_to_rpolars_err)
             .map(DataFrame)
     }
+
+    pub fn write_csv(
+        &mut self,
+        path: Robj,
+        has_header: Robj,
+        separator: Robj,
+        line_terminator: Robj,
+        quote: Robj,
+        batch_size: Robj,
+        datetime_format: Robj,
+        date_format: Robj,
+        time_format: Robj,
+        float_precision: Robj,
+        null_value: Robj,
+        quote_style: Robj,
+    ) -> RResult<()> {
+
+        let null = robj_to!(String, null_value).unwrap();
+        let path = robj_to!(str, path).unwrap();
+        let f = std::fs::File::create(path).unwrap();
+        let qs = parse_quote_style(quote_style);
+
+        CsvWriter::new(f)
+            .has_header(robj_to!(bool, has_header).unwrap())
+            .with_delimiter(robj_to!(u8, separator).unwrap())
+            .with_line_terminator(robj_to!(String, line_terminator).unwrap())
+            .with_quoting_char(robj_to!(u8, quote).unwrap())
+            .with_batch_size(robj_to!(usize, batch_size).unwrap())
+            .with_datetime_format(robj_to!(Option, String, datetime_format).unwrap())
+            .with_date_format(robj_to!(Option, String, date_format).unwrap())
+            .with_time_format(robj_to!(Option, String, time_format).unwrap())
+            .with_float_precision(robj_to!(Option, usize, float_precision).unwrap())
+            .with_null_value(null)
+            .with_quote_style(qs)
+            .finish(&mut self.0)
+            .map_err(polars_to_rpolars_err)
+    }
 }
+
+pub fn parse_quote_style(x: Robj) -> QuoteStyle {
+    match robj_to!(Option, String, x).unwrap_or_default().unwrap().as_str() {
```

Comment:
I would change .unwrap_or_default().unwrap().as_str() to ?.as_str()

wrap Ok(QuoteStyle::Always) and other lines

not use panic! nor unreachable as a use could supply a "wrong_name"

instead use rerr().plain("polars internal error: `quote_style` must be 'always', 'necessary' or 'non_numeric'.") which returns Err(rpolarserr) which is same type as Ok(QuoteStyle) being  RResult<QuotoStyle>.

Then after match arms use ? again to get bubble the error and only work with ok value